seed/seed20.go

```diff
@@ -96,32 +94,74 @@ type seed20 struct {
 	essentialSnapsNum int
 }
 
-// Copy implement Copier interface.
-func (s *seed20) Copy(seedDir string, label string, tm timings.Measurer) (err error) {
+func shouldCopySnap(target *Snap, model *asserts.Model, oc *OptionalContainers) bool {
+	if oc == nil {
+		return true
+	}
+
+	if target.Essential {
+		return true
+	}
+
+	for _, sn := range model.AllSnaps() {
+		if sn.Name != target.SnapName() {
+			continue
+		}
+
+		if sn.Presence == "required" {
+			return true
```

Comment:
tests don't seem to exercise this case

Reply:
Added in cea09b74b0ed4cd1e1c6e3944b6252bf0f54a362.